types/validator_set.go

```diff
@@ -953,3 +963,33 @@ func safeSubClip(a, b int64) int64 {
 	}
 	return c
 }
+
+func safeMul(a, b int64) (int64, bool) {
+	if a == 0 || b == 0 {
+		return 0, false
+	}
+
+	absOfB := b
+	if b < 0 {
+		absOfB = -b
+	}
+
+	var (
+		c        = a
+		overflow bool
+	)
+
+	for absOfB > 1 {
+		c, overflow = safeAdd(c, a)
+		if overflow {
+			return c, true
+		}
+		absOfB--
+	}
+
+	if (b < 0 && a > 0) || (b < 0 && a < 0) {
```

Comment:
Can't this just be `if b < 0` ?

Reply:
no. `b < 0 && a != 0` should work though